src/OpenTelemetry.Instrumentation.StackExchangeRedis/Implementation/RedisProfilerEntryInstrumenter.cs

```diff
@@ -88,30 +95,37 @@ static bool GetCommandAndKey(
             StackExchangeRedisConnectionInstrumentation.CreationTags,
             startTime: command.CommandCreated);
 
-        if (activity == null)
+        if (activity is null && metrics.Enabled is false)
         {
             return null;
         }
 
-        activity.SetEndTime(command.CommandCreated + command.ElapsedTime);
+        activity?.SetEndTime(command.CommandCreated + command.ElapsedTime);
+        var meterTags = metrics.Enabled ?
+            new TagList(StackExchangeRedisConnectionInstrumentation.CreationTags) :
+            default(IList<KeyValuePair<string, object?>>);
 
-        if (activity.IsAllDataRequested)
-        {
-            // see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md
+        // see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md
+
+        // Timing example:
+        // command.CommandCreated; //2019-01-10 22:18:28Z
 
-            // Timing example:
-            // command.CommandCreated; //2019-01-10 22:18:28Z
+        // command.CreationToEnqueued;      // 00:00:32.4571995
+        // command.EnqueuedToSending;       // 00:00:00.0352838
+        // command.SentToResponse;          // 00:00:00.0060586
+        // command.ResponseToCompletion;    // 00:00:00.0002601
 
-            // command.CreationToEnqueued;      // 00:00:32.4571995
-            // command.EnqueuedToSending;       // 00:00:00.0352838
-            // command.SentToResponse;          // 00:00:00.0060586
-            // command.ResponseToCompletion;    // 00:00:00.0002601
+        // Total:
+        // command.ElapsedTime;             // 00:00:32.4988020
 
-            // Total:
-            // command.ElapsedTime;             // 00:00:32.4988020
+        var flags = command.Flags.ToString();
+        activity?.SetTag(SemanticConventions.AttributeDbRedisFlagsKeyName, flags);
 
-            activity.SetTag(StackExchangeRedisConnectionInstrumentation.RedisFlagsKeyName, command.Flags.ToString());
+        var operationName = command.Command ?? string.Empty;
+        meterTags?.Add(SemanticConventions.AttributeDbOperationName, operationName);
```

Comment:
`AttributeDbOperationName` was never set for Activity though that is required there. Need to check why was that the case before adding it to Metrics.

Reply:
Here we are creating the metrics following the most recent semantic conventions, on the other hand, tracing still follows an old version. I think it's better to keep the activities as is and handle it in a separate PR, as you suggested here: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1982#pullrequestreview -2251326101